### DISCUSSION TITLE
Text style not applying unless property is defined

### DISCUSSION THREAD

**Discussion by forrest-keller** [chakra-ui/chakra-ui#3612] [Q&A]
I am running into an issue where if I define a textStyle as follows:

```
h1: {
    fontSize: "100px"
}
```

It works, but only when I define the fontSize property on my component as such:

`<Heading fontSize="100px" textStyle="h1">Test</Heading>`

Any tips on why this might be happening? Everything else seems to be working all right.

**Comment by malashevskyi** [comment]
Because chakra has default styles for Heading component and it overrides your styles.

"I am running into an issue where if I define a textStyle as follows:" Where do you define your styles?

**Reply by spidertwin2** [reply]
Hi,

In theme.ts, I am unable to customise `fontWeight` and `fontSize`. Other props are working fine.

```
import { extendTheme } from "@chakra-ui/react"

const customTheme = extendTheme({
  textStyles: {
      pageSubHeading: {
        fontSize: ["md", "l", "xl"],
        letterSpacing: "0.5px",
        color: "gray.400",
        fontWeight: "300",
      },
    }
})

export default customTheme
```
:cactus:

**Reply by spidertwin2** [reply]
I just realized textStyle does not work on `<Heading>`.  Changed it to `<Text>` and then it works.

**Comment by jcespinoza** [comment]
Borrowing from this conversation, in my case I managed to make it respect my fontSize and fontWeight by defining them with dashes on the theme literal like so:
```
{
  textStyles: {
    formLabels: {
      'font-size': '16px',
      'font-weight': '400',
      lineHeight: '14px'
    }
  }
}
```

Strangely enough, `lineHeight' was respected without this hack.

**Reply by primos63** [reply]
These are the theme properties for form-label
```javascript
const baseStyle: SystemStyleObject = {
  fontSize: "md",
  marginEnd: 3,
  mb: 2,
  fontWeight: "medium",
  transitionProperty: "common",
  transitionDuration: "normal",
  opacity: 1,
  _disabled: {
    opacity: 0.4,
  },
}
```
Your textStyle works because for the formLabels `fontSize` and not `font-size` is defined so the match fails (same for font-weight). You didn't have to do that for lineHeight because it isn't defined in the base style.
I have a [PR for this issue](https://github.com/chakra-ui/chakra-ui/pull/4736). It just hasn't gone through yet.
